File: sync/ttfl.py
"""TTFL score calculator.

Formula: (PTS + REB + AST + STL + BLK + FGM + 3PM + FTM)
       - (TOV + FG_miss + 3P_miss + FT_miss)
"""


def compute_ttfl_score(
    pts: int, reb: int, ast: int, stl: int, blk: int,
    fgm: int, fga: int, tpm: int, tpa: int, ftm: int, fta: int, tov: int,
) -> int:
    positive = pts + reb + ast + stl + blk + fgm + tpm + ftm
    negative = tov + (fga - fgm) + (tpa - tpm) + (fta - ftm)
    return positive - negative


def compute_ttfl_from_game_log(log: dict) -> int:
    """Compute TTFL score from an nba_api game log dict."""
    return compute_ttfl_score(
        pts=log["PTS"], reb=log["REB"], ast=log["AST"],
        stl=log["STL"], blk=log["BLK"],
        fgm=log["FGM"], fga=log["FGA"],
        tpm=log["FG3M"], tpa=log["FG3A"],
        ftm=log["FTM"], fta=log["FTA"],
        tov=log["TOV"],
    )

File: web/src/lib/ttfl.ts
export function computeTtflScore(stats: {
  pts: number;
  reb: number;
  ast: number;
  stl: number;
  blk: number;
  fgm: number;
  fga: number;
  tpm: number;
  tpa: number;
  ftm: number;
  fta: number;
  tov: number;
}): number {
  const positive =
    stats.pts +
    stats.reb +
    stats.ast +
    stats.stl +
    stats.blk +
    stats.fgm +
    stats.tpm +
    stats.ftm;
  const negative =
    stats.tov +
    (stats.fga - stats.fgm) +
    (stats.tpa - stats.tpm) +
    (stats.fta - stats.ftm);
  return positive - negative;
}
